### PR TITLE
Allow application/geo+json response types

### PR DESCRIPTION
**Pull request by olof-nord**
Some well defined apis define their geojson types in OpenAPI specs, and those may be of the type `application/geo+json`: https://datatracker.ietf.org/doc/html/rfc7946. 

This include some of the API specs I am using, which when using this type today to render TS oazapfts unfortunately ends up using the Blob data type.

This PR adds support for this content type just as it has for `application/hal+json` and `application/problem+json`.

In short, this PR is based on the recently merged https://github.com/cellular/oazapfts/pull/197 one.

### CODE REVIEW

**Review by shocklateboy92 — CHANGES_REQUESTED** [review]
Hi @olof-nord and thanks for your PR!

The change seems sensible, but could you please add a test that covers the new behavior.

**Comment by jonkoops** [comment]
I'm starting to think that we should perhaps change this to some kind of regular expression that checks the mime type for `application/*+json`, or perhaps provide an option to pass a function that allows the mime-type to be checked in user-space.

**Review by shocklateboy92 — APPROVED** [review]
I'm actually okay with this. We can do a regex if the cases get too unwieldy.
Also happy this increases our test coverage (I think) :test_tube: 

However, I'll give a few days before merging so @jonkoops can chime in if he feels strongly about the regex.

**Review by jonkoops — APPROVED** [review]
LGTM

**Comment by jonkoops** [comment]
I think extending this for now is more than fine, but might get unwieldy as we go.

**Comment by olof-nord** [comment]
Thanks for the good feedback, always happy to increase that test coverage 👍
